flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNITCase.java

```diff
@@ -64,32 +69,46 @@
 	@BeforeClass
 	public static void setup() {
 		YARN_CONFIGURATION.set(YarnTestBase.TEST_CLUSTER_NAME_KEY, "flink-yarn-tests-per-job");
-		startYARNWithConfig(YARN_CONFIGURATION);
+		startYARNWithConfig(YARN_CONFIGURATION, true);
```

Comment:
I'd suggest you just create a new file and write class about yarn file replication, without any modification on `YARNITCase`. It doesn't no harm we add a new file(test case) and if we keep modifying existing file for "patching" logics we highly possibly run into mess.

Besides, still you don't test 

```
configuration.setInteger(YarnConfigOptions.FILE_REPLICATION, 4);
```

with value expect `4` especially edge case such as `< -1` `== -1`(possibly `<= -1` is ok) `== 0` `== 1`.

Reply:
A reason that I suggest add a new file is that previously we test `YARNITCase` with `withDFS == false` but now they are all `true`. I don't see an obvious reason to do it and we possibly loss test coverage.

Reply:
From the perspective of long term maintenance, I agree with your point. I created another test class exends from YARNITCase. It will not touch existing test cases.